### ISSUE TITLE
[SUGGESTION] Allow Method groups as a substitute for an BsonExpression

### ISSUE THREAD

**Issue by agaertner** [litedb-org/LiteDB#2186]
**Is your feature request related to a problem? Please describe.**
Code gets bulky and inability to use method groups as a substitute for expressions causes duplicated expressions ie. code fragments.

**Describe the solution you'd like**
Overload that translated my static method group
```csharp
db.FindOne(MyModel.IsValid)
```
to a BsonExpression equivalent which is:
```csharp
db.FindOne(x => x.Foobar && x.Arbitrary > 0)
```
where MyClass.IsValid in the example is defined as follows.
```csharp
public class MyModel {
    public bool Foobar { get; set; }
    public long Arbitrary { get; set; }
    public static bool IsValid(MyModel model) => model.Foobar && model.Arbitrary > 0;
}
```

**Describe alternatives you've considered**
Dublicated code fragments (expressions), but code gets bulky (see above) because dublicated expression cannot be managed from a single place and are spread across the project.

**Comment by blackholeearth** [comment]
bro i tested that, if you remove the static keyword you can do it like this ??!!

```
public class MyModel {
    public bool Foobar { get; set; }
    public long Arbitrary { get; set; }
    public bool IsValid => Foobar && Arbitrary > 0;
}

// in code
            var resultsxx = col.Find(x => x.isValid ) ;

```

i didnt encount any syntax error .

is there a reason force you to make it static ??

**Comment by agaertner** [comment]
Of course. But this is not the same syntactic sugar I am requesting. I specifically request to have method groups be passable without ```x => MyModel.IsValid(x)``` when the LINQ is expecting a single parametrized method.

**Comment by JKamsker** [comment]
Thanks for the suggestion! It's now part of #2834, which collects the related feature requests in one place so they can be discussed and prioritised together. Closing this as a duplicate — please follow #2834 and add your use case there.
